lib/miq_expression.rb

```diff
@@ -1554,36 +1554,43 @@ def self._quick_search?(e)
     end
   end
 
-  def custom_attribute_columns_and_models(expression = nil)
-    return custom_attribute_columns_and_models(exp).uniq if expression.nil?
+  def custom_attribute_columns_and_models
+    @virtual_custom_attributes ||= begin
+      @virtual_custom_attributes = []
+      _custom_attribute_columns_and_models
+      @virtual_custom_attributes
+    end
+  end
+
+  def _custom_attribute_columns_and_models(expression = nil)
```

Comment:
Could we make this private and lose the `_` prefix? With perhaps a more intention-revealing name?